qmlbind/include/qmlbind/qmlbind_global.h

```diff
@@ -30,49 +39,101 @@ class SignalEmitter;
+typedef QSharedPointer<QmlBind::Interface> qmlbind_interface;
+typedef QSharedPointer<QmlBind::MetaObject> qmlbind_metaobject;
+typedef QSharedPointer<QmlBind::Exporter> qmlbind_exporter;
 
-typedef QmlBind::SignalEmitter *qmlbind_signal_emitter;
+typedef QmlBind::SignalEmitter qmlbind_signal_emitter;
 
 #else
 
-typedef struct qmlbind_application_s {} *qmlbind_application;
+typedef struct qmlbind_application {} qmlbind_application;
 
-typedef struct qmlbind_engine_s {} *qmlbind_engine;
-typedef struct qmlbind_component_s {} *qmlbind_component;
-typedef struct qmlbind_plugin_s {} *qmlbind_plugin;
+typedef struct qmlbind_engine {} qmlbind_engine;
+typedef struct qmlbind_component {} qmlbind_component;
+typedef struct qmlbind_plugin {} qmlbind_plugin;
 
-typedef struct qmlbind_value_s {} *qmlbind_value;
-typedef struct qmlbind_iterator_s {} *qmlbind_iterator;
-typedef struct qmlbind_string_s {} *qmlbind_string;
+typedef struct qmlbind_value {} qmlbind_value;
+typedef struct qmlbind_iterator {} qmlbind_iterator;
+typedef struct qmlbind_string {} qmlbind_string;
 
-typedef struct qmlbind_interface_s {} *qmlbind_interface;
-typedef struct qmlbind_metaobject_s {} *qmlbind_metaobject;
-typedef struct qmlbind_exporter_s {} *qmlbind_exporter;
+typedef struct qmlbind_interface {} qmlbind_interface;
+typedef struct qmlbind_metaobject {} qmlbind_metaobject;
+typedef struct qmlbind_exporter {} qmlbind_exporter;
 
-typedef struct qmlbind_signal_emitter_s {} *qmlbind_signal_emitter;
+typedef struct qmlbind_signal_emitter {} qmlbind_signal_emitter;
 
 #endif
 
-typedef struct qmlbind_backref_s *qmlbind_backref;
+typedef struct qmlbind_backref {} qmlbind_backref;
+
+
+/*!
+ * \brief encapsulates functions that are used from QML for C object creation & destruction,
+ * method calls and property modification.
+ *
+ * Those functions need to be defined by libqmlbind's users.
+ *
+ * This struct is needed to construct new `qmlbind_interface`s. All function pointers need to be defined, a nullptr is
+ * not a valid argument.
+ */
+typedef struct qmlbind_interface_handlers {
+    /*!
+     * \brief creates a new C object to be used in QML.
+     *
+     * \param classRef the `qmlbind_backref` that was given to `qmlbind_exporter_new`. Does not transfer ownership.
+     * \param signalEmitter should be stored in the returned object to be able to emit signals. Ownership is transfered.
+     * \returns the newly-created object, transfering ownership.
+     */
+    qmlbind_backref* (*new_object)(qmlbind_backref *classRef, qmlbind_signal_emitter *signalEmitter);
+
+    /*!
+     * \brief deletes a C object that was created by `new_object`.
+     */
+    void (*delete_object)(qmlbind_backref *objRef);
+
+    /*!
+     * \brief executes the `name` method defined on the `objRef` C object, with `argc` parameters in `argv`, and returns
+     * the result.
+     *
+     * This method must take ownership of the `qmlbind_values` in the `argv` array.
```

Comment:
My fault, I updated the code but forgot to update the comments. :laughing: Good that you found it.